src/ucs/sys/ucs_netlink.c

```diff
@@ -0,0 +1,96 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "ucs_netlink.h"
+
+#include <ucs/sys/sock.h>
+#include <ucs/sys/compiler.h>
+#include <ucs/type/status.h>
+#include <ucs/debug/log.h>
+
+#include <errno.h>
+#include <linux/netlink.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+
+static ucs_status_t ucs_netlink_socket_create(int *fd, int protocol)
+{
+    ucs_status_t ret;
+    struct sockaddr_nl sa = {0};
+
+    ret = ucs_socket_create(AF_NETLINK, SOCK_RAW, protocol, fd);
+    if (ret != UCS_OK) {
+        ucs_diag("failed to create netlink socket %d", ret);
+        *fd = -1;
+        return ret;
+    }
+
+    sa.nl_family = AF_NETLINK;
+
+    if (bind(*fd, (struct sockaddr*)&sa, sizeof(sa)) < 0) {
+        ucs_close_fd(fd);
+        ucs_diag("failed to bind netlink socket %d", *fd);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    return UCS_OK;
+}
+
+ucs_status_t
+ucs_netlink_send_cmd(int protocol, void *nl_protocol_hdr,
+                     size_t nl_protocol_hdr_size, char *recv_msg_buf,
+                     size_t *recv_msg_buf_len, unsigned short nlmsg_type)
+{
+    ucs_status_t ret;
+    int fd;
+    struct nlmsghdr nlh;
+    struct iovec iov[2];
+    size_t send_msg_len = NLMSG_LENGTH(nl_protocol_hdr_size);
+
+    ret = ucs_netlink_socket_create(&fd, NETLINK_ROUTE);
+    if (ret != UCS_OK) {
+        ucs_diag("failed to open netlink socket");
+        return ret;
+    }
+
+    memset(iov, 0, sizeof(iov));
+    nlh.nlmsg_len   = send_msg_len;
```

Comment:
`nlh.nlmsg_len = NLMSG_LENGTH(nl_protocol_hdr_size);`

Reply:
I wanted to use another variable so I could send its pointer to `ucs_socket_sendv_nb`. I know can technically send `&nlh.nlmsg_len` but for some reason it feels unsafe.. is there a reason for that or should I use `&nlh.nlmsg_len`?